beam-postgres/Database/Beam/Postgres/Connection.hs

```
@@ -198,9 +199,14 @@ withPgDebug dbg conn (Pg action) =
              case action' of
                 PgStreamDone (Right x) -> do
                   start <- getTime Monotonic
-                  Pg.execute_ conn (Pg.Query query)
-                  end <- getTime Monotonic
-                  (, Just (end - start)) <$> next x
+                  (respWithException ::Either SomeException a) <- try $ Pg.execute_ conn (Pg.Query query)
+                  case respWithException of
+                    Left err -> do
+                      dbg (decodeUtf8 query)
+                      return (Left $ BeamRowReadError Nothing $ ColumnErrorInternal (show err ) , Nothing) 
```

We can return the same exception that we used for case match